# Lab 2.03 - Univariate Analysis


In [2]:
# Importing the necessary packages
import numpy as np                                  # "Scientific computing"
import scipy.stats as stats                         # Statistical tests

import pandas as pd                                 # Data Frame
from pandas.api.types import CategoricalDtype

import matplotlib.pyplot as plt                     # Basic visualisation
from statsmodels.graphics.mosaicplot import mosaic  # Mosaic diagram
import seaborn as sns                               # Advanced data visualisation

## Exercise 3 - Kitten litters

A vet keeps a record of the number of kittens in litters produced by cats in his practice. The code block below contains the data as a frequency table.

- First put the data into one list of individual observations: [1, 1, 2, 2, ... 5, 5, 6, 6, 6, 6, 7, 7, 8]  
- Visualise the data set using all suitable chart types that you know of.
- Calculate:
    - the median size of a litter
    - the mean size of a litter
    - the lower quartile
    - the upper quartile

In [3]:
kittens = pd.DataFrame.from_dict({
    'Number in litter': [1, 2, 3,  4, 5, 6, 7, 8],
    'Frequency':        [2, 4, 7, 11, 8, 4, 2, 1]})
kittens

Number in litter  Frequency
0                 1          2
1                 2          4
2                 3          7
3                 4         11
4                 5          8
5                 6          4
6                 7          2
7                 8          1

In [63]:
a = []
for x in range(len(kittens)):
    a = a + ([kittens['Number in litter'][x]] * kittens['Frequency'][x]) 

m = np.mean(a)
media = np.median(a)
quantile = np.quantile(a, [.25, .75])

print(m)
print(media)
print(quantile)
print(a)

4.128205128205129
4.0
[3. 5.]
[1, 1, 2, 2, 2, 2, 3, 3, 3, 3, 3, 3, 3, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 5, 5, 5, 5, 5, 5, 5, 5, 6, 6, 6, 6, 7, 7, 8]


### Some answers:

- Median: 4.0
- Mean: 4.128205128205129
- Lower Quartile: 3.0
- Upper Quartile: 5.0

Source: Complete Probability & Statistics 1 for Cambridge International As & A Level